library/kotlin/src/io/envoyproxy/envoymobile/RequestMapper.kt

```diff
@@ -4,13 +4,17 @@ internal fun Request.outboundHeaders(): Map<String, List<String>> {
   val retryPolicyHeaders = retryPolicy?.outboundHeaders() ?: emptyMap()
 
   val result = mutableMapOf<String, List<String>>()
-  result.putAll(headers.filter { entry -> !entry.key.startsWith(":") })
+  result.putAll(headers.filter { entry -> !(entry.key.startsWith(":") || entry.key.startsWith("x-envoy-mobile"))})
```

Comment:
Up to you but using this might be more readable: `!entry.key.startsWith(":") && !entry.key.startsWith("x-envoy-mobile")`

Reply:
I am always ambivalent about which one is more readable. I'll change to kick CI... 💩

Reply:
😭